src/services/import/excel/XLSImporterValidateObservations.ts

```diff
@@ -0,0 +1,131 @@
+import { NextFunction, Request, Response } from 'express';
+import Excel, { Workbook } from 'exceljs';
+import { getRepository, Repository } from 'typeorm';
+import {
+  checkObservationsHeaderNames,
+  checkObservationImportedData,
+  HeaderCheck,
+  DataCheck,
+  EURingError,
+  RowErorr,
+  RowValidatedData,
+  RawData,
+} from './helper';
+import { MulterOptions } from '../../../controllers/upload-files-controller';
+import AbstractImporter, { ImporterType } from '../AbstractImporter';
+import { Species } from '../../../entities/euring-codes/species-entity';
+import { Sex } from '../../../entities/euring-codes/sex-entity';
+import { Age } from '../../../entities/euring-codes/age-entity';
+import { Status } from '../../../entities/euring-codes/status-entity';
+import { CustomError } from '../../../utils/CustomError';
+
+export default class XLSImporterValidateObservations extends AbstractImporter {
+  public type: ImporterType = 'VALIDATE-XLS';
+
+  public route: string = 'observations';
+
+  public options: MulterOptions = {
+    extensions: ['.xls', '.xlsx'],
+    any: true,
+  };
+
+  private status: Repository<Status> = getRepository(Status);
+
+  private species: Repository<Species> = getRepository(Species);
+
+  private sex: Repository<Sex> = getRepository(Sex);
+
+  private age: Repository<Age> = getRepository(Age);
+
+  private checkEuRingCodes = async (excelData: DataCheck): Promise<void> => {
+    try {
+      /* eslint-disable */
+      for (const row of excelData.validFormatData) {
+        const { data, rowNumber }: RowValidatedData = row;
+        const { eu_statusCode, eu_species, eu_sexCode, eu_ageCode }: RawData = data;
+
+        if (data) {
```

Comment:
it's too expensive. eahc row produce 4 requests then multipy by row's amout
Suggets to use here chached repository as first solution